crates/pop-cli/src/main.rs

```diff
@@ -38,41 +40,126 @@ enum Commands {
 	Test(commands::test::TestArgs),
 }
 
+fn init_config() -> Result<()> {
+	match pop_telemetry::write_default_config() {
+		Ok(maybe_path) => {
+			if let Some(path) = maybe_path {
+				log::info(format!("Initialized config file at {}", &path.to_str().unwrap()))?;
+			}
+		},
+		Err(err) => {
+			log::warning(format!(
+				"Unable to initialize config file, continuing... {}",
+				err.to_string()
+			))?;
+		},
+	}
+	Ok(())
+}
+
 #[tokio::main]
 async fn main() -> Result<()> {
+	init_config()?;
+
+	// handle for await not used here as telemetry should complete before any of the commands do.
+	tokio::spawn(pop_telemetry::record_cli_used());
+
+	// If error occurs, this will be used to ensure error telemetry is complete before destructing.
+	// This await handle is used as the error will not have sufficient time to report before destruction.
+	let mut tel_error_handle: Option<JoinHandle<pop_telemetry::Result<()>>> = None;
+
 	let cli = Cli::parse();
-	match cli.command {
+	let res = match cli.command {
 		Commands::New(args) => Ok(match &args.command {
 			#[cfg(feature = "parachain")]
-			commands::new::NewCommands::Parachain(cmd) => cmd.execute().await?,
+			commands::new::NewCommands::Parachain(cmd) => cmd.execute().await.map_err(|err| {
```

Comment:
Might be a good opportunity to clean this up by importing the commands to eliminate the commands::* prefixes.